plugins/extensions/plugins/ZeroEQPlugin.cpp

```diff
@@ -179,7 +189,47 @@ void ZeroEQPlugin::_materialUpdated( )
         scene->commitMaterials( true );
         _extensionParameters.engine->getFrameBuffer()->clear();
     }
+}
 
+bool ZeroEQPlugin::_requestTransferFunction1D()
+{
+    ScenePtr scene = _extensionParameters.engine->getScene();
+    TransferFunction& transferFunction = scene->getTransferFunction();
+    std::vector< ::zerobuf::render::Point2D > items;
+
+    for( const auto& controlPoint: transferFunction.getControlPoints( TF_RED ) )
+        items.push_back( ::zerobuf::render::Point2D(
+            controlPoint.x(), controlPoint.y() ));
+    _remoteTransferFunction1D.setAttribute( transferFunction.getAttributeAsString( TF_RED ) );
+    _remoteTransferFunction1D.setPoints( items );
+    return true;
+}
+
+void ZeroEQPlugin::_transferFunction1DUpdated( )
+{
+    ScenePtr scene = _extensionParameters.engine->getScene();
+    TransferFunction& transferFunction = scene->getTransferFunction();
+    std::vector< ::zerobuf::render::Point2D > points = _remoteTransferFunction1D.getPointsVector();
+
+    const std::string& attributeName = _remoteTransferFunction1D.getAttributeString();
+    BRAYNS_INFO << "Setting "
+                << points.size() << " control points for transfer function attribute <"
+                << attributeName <<  ">" << std::endl;
+
+    const TransferFunctionAttribute attribute =
+        transferFunction.getAttributeFromString( attributeName );
+
+    if( attribute != TF_UNDEFINED )
```

Comment:
if( attribute == TF_UNDEFINED )
   return; 

is nicer IMHO but that is your call.